modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1486,9 +1486,13 @@ def astype(self, col_dtypes, errors: str = "raise"):
         BaseDataFrame
             Dataframe with updated dtypes.
         """
+        self_dtypes = self.dtypes
+        if all(is_dtype_equal(self_dtypes[k], v) for k, v in col_dtypes.items()):
+            return self
```

Comment:
should probably return a copy here, so in-place modification of the result wouldn't propagate the original frame. All copies are shallow, so it shouldn't be expensive
```suggestion
            return self.copy()
```

Reply:
or maybe just move this `is_dtype_equal()` comparison [to the inside of the for-loop](https://github.com/modin-project/modin/pull/6662/files#diff-718e1f98ecbed40356292a477fc27bafb81e62c796491c81e398d5b397d6d69eR1505), and then after the loop do?:
```python
if not any_dtype_has_changed:
   return self.copy()
```